wavesrv/pkg/cmdrunner/cmdrunner.go

```diff
@@ -4221,6 +4243,72 @@ func MarkdownViewCommand(ctx context.Context, pk *scpacket.FeCommandPacketType)
 	return update, nil
 }
 
+func StatDir(ctx context.Context, ids resolvedIds, path string, fileCallback func(pk *packet.FileStatPacketType, done bool, err error)) {
+	statPk, _, err := getFileStat(ctx, ids, path)
+	if err != nil {
+		fileCallback(nil, true, err)
+		return
+	}
+	if !statPk.IsDir {
+		fileCallback(statPk, true, nil)
+		return
+	} else {
+		cwd := ids.Remote.FeState["cwd"]
+		listDirPk := packet.MakeListDirPacket()
+		listDirPk.ReqId = uuid.New().String()
+		if filepath.IsAbs(path) {
+			listDirPk.Path = path
+		} else {
+			listDirPk.Path = filepath.Join(cwd, path)
+		}
+		msh := ids.Remote.MShell
+		listDirIter, err := msh.ListDir(ctx, listDirPk)
+		if err != nil {
+			fileCallback(nil, true, err)
+			return
+		}
+		log.Printf("listDirIter: %v", listDirIter)
+	}
+}
+
+func FileViewCommand(ctx context.Context, pk *scpacket.FeCommandPacketType) (sstore.UpdatePacket, error) {
+	ids, err := resolveUiIds(ctx, pk, R_Session|R_Screen|R_RemoteConnected)
+	path := ids.Remote.FeState["cwd"]
+	if len(pk.Args) > 0 {
+		path = pk.Args[0]
+	}
+	log.Printf("path: %v", path)
+	if err != nil {
+		return nil, err
+	}
+	cmd, err := makeStaticCmd(ctx, GetCmdStr(pk), ids, pk.GetRawStr(), nil)
+	if err != nil {
+		return nil, err
+	}
+	lineState := make(map[string]any)
+	update, err := addLineForCmd(ctx, "/"+GetCmdStr(pk), false, ids, cmd, "fileview", lineState)
+	if err != nil {
+		return nil, err
+	}
+	update.Interactive = pk.Interactive
+	sstore.MainBus.SendScreenUpdate(cmd.ScreenId, update)
+
+	go func() {
+		var outputPos int64
+		StatDir(context.Background(), ids, path, func(statPk *packet.FileStatPacketType, done bool, err error) {
```

Comment:
callback paradigm, thoughts? 
I thought this was a little less code than using a channel here 
and I think it works just as well because the callback should be a closure on the data that we need like the context and the cmd object etc

Reply:
I guess the down side is the execution of the StatDir function stops while the callback is being executed
I think that is okay for this purpose because the callback is just gonna write to the pty and then return

but maybe a channel would make it faster